### PR TITLE
fix(sdk): Always create output paths for parameters in addition to artifacts in v2 Python components.

### PR DESCRIPTION
**Pull request by neuromage**
This PR also mirrors useful dsl constructs under kfp.v2.dsl to streamline imports
for end users (it was meant to be a different PR but ended up being committed while fixing this).

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

### CODE REVIEW

File: sdk/python/kfp/v2/dsl/__init__.py [hunk 1]
```diff
@@ -13,6 +13,19 @@
 # limitations under the License.
 
 from kfp.v2.dsl.component_decorator import component
+from kfp.dsl.io_types import (
```
Comment (neuromage) [inline sdk/python/kfp/v2/dsl/__init__.py:16]: I can separate this into a different PR if that's preferable.

Comment (chensun) [reply sdk/python/kfp/v2/dsl/__init__.py:16]: That's fine.

**Review by chensun — COMMENTED** [review]
/lgtm
/approve

**Comment by amygdala** [comment]
Hmm, I'm getting behaviour that suggests this isn't fixed (see my internal chat thread)

**Pull request merged**